lib/dialects/mariadb/query-generator.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const _ = require('lodash');
+const Utils = require('../../utils');
+const MySQLQueryGenerator = require('../mysql/query-generator');
+const util = require('util');
+
+class MariaDBQueryGenerator extends MySQLQueryGenerator {
+  constructor(options) {
+    super(options);
+  }
+
+  createSchema(schema, options) {
+    options = Object.assign({
+      charset: null,
+      collate: null
+    }, options || {});
+
+    const charset = options.charset ? ` DEFAULT CHARACTER SET ${this.escape(options.charset)}` : '';
+    const collate = options.collate ? ` DEFAULT COLLATE ${this.escape(options.collate)}` : '';
+
+    return `CREATE SCHEMA IF NOT EXISTS ${this.quoteIdentifier(schema)}${charset}${collate};`;
+  }
+
+  dropSchema(schema) {
+    return `DROP SCHEMA IF EXISTS ${this.quoteIdentifier(schema)};`;
+  }
+
+  showSchemasQuery(options) {
+    const skip =  options.skip && Array.isArray(options.skip) && options.skip.length > 0 ? options.skip : null;
+    return `SELECT SCHEMA_NAME as schema_name FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME NOT IN ('MYSQL', 'INFORMATION_SCHEMA', 'PERFORMANCE_SCHEMA'${skip ? skip.reduce( (sql, schemaName) => sql +=  `,${this.escape(schemaName)}`, '') : ''});`;
+  }
+
+  showTablesQuery() {
+    return 'SELECT TABLE_NAME, TABLE_SCHEMA FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA NOT IN (\'MYSQL\', \'INFORMATION_SCHEMA\', \'PERFORMANCE_SCHEMA\') AND TABLE_TYPE = \'BASE TABLE\'';
+  }
+
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = this.parseConditionObject(smth.conditions).map(
+          condition =>
+            `json_unquote(json_extract(${this.quoteIdentifier(
+              condition.path[0])},'\$.${_.tail(condition.path).join(
+              '.')}')) = '${condition.value}'`
+        );
+
+        return conditions.join(' and ');
+      }
+      if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the sqlite json functions
+        if (this._checkValidJsonStatement(smth.path)) {
+          str = smth.path;
+        } else {
+          // Also support json dot notation
+          let path = smth.path;
+          let startWithDot = true;
+
+          // Convert .number. to [number].
+          path = path.replace(/\.(\d+)\./g, '[$1].');
```

Comment:
I think you only need to bottom one?

Reply:
it depends if you want to support JSON dot notation or not. 

```sql
SET @json = '{"tt": [1, 2, [3, 4]], "tt2" : 3}';
SELECT JSON_EXTRACT(@json, '$.tt.2'); // return null;
SELECT JSON_EXTRACT(@json, '$.tt[2]'); // return [3,4];
SELECT JSON_EXTRACT(@json, '$.tt[2].1'); // return null;
SELECT JSON_EXTRACT(@json, '$.tt[2][1]'); // return 4;
```
so supporting something like whereItemQuery(Sequelize.json('profile.id.0.1'), '1')
`profile.id.0.1` needs to be changed to json_unquote(json_extract(`profile`,'$.id[0][1]')) = '1'